app/src/main/java/uk/gov/android/securestore/SharedPrefsStore.kt

```diff
@@ -1,42 +1,63 @@
 package uk.gov.android.securestore
 
 import android.content.Context
+import androidx.fragment.app.FragmentActivity
 import java.security.GeneralSecurityException
 import java.security.KeyStoreException
+import kotlin.coroutines.suspendCoroutine
+import uk.gov.android.securestore.authentication.AuthenticatorPromptConfiguration
 import uk.gov.android.securestore.crypto.CryptoManager
 import uk.gov.android.securestore.crypto.RsaCryptoManager
 
 class SharedPrefsStore(
-    context: Context,
+    context: FragmentActivity,
```

Comment:
Ah it's because of the android `BiometricPrompt` in `RsaCryptoManager` / `UserAuthentication`.

We able to pass these in as function parameters somewhere so that the building of those objects stays within the Fragment / Activity code? It means we can pass around the interfaces we control (`Authenticator` / `CryptoManager`) instead of building it a couple of layers away?

Reply:
Yep, moved the authenticator to being built in the sharedStore consturctor. They have default values but the calling Activity/Fragment can choose to create it's own `CryptoManager` or `Authenticator` now